tests/functional/object/rgw/test_rgw_http_endpoint.py

```
@@ -0,0 +1,298 @@
+                    "which indicates the endpoint is effectively inaccessible"
+                )
+                log.info("Step 5 passed: HTTP endpoint is not accessible")
+            else:
+                log.error(
+                    f"HTTP endpoint returned success status {response.status_code} "
+                    "even though disableHttp is true - this is unexpected!"
+                )
+                assert False, (
+                    "HTTP endpoint should not be accessible when disableHttp is true, "
+                    f"but got response with status code {response.status_code}"
+                )
```

Ruff (0.15.14)

[warning] Assertion always fails, replace with `pytest.fail()`

(PT015)